rotkehlchen/globaldb/upgrades/v7_v8.py

```diff
@@ -66,11 +66,54 @@ def fix_detected_spam_tokens(write_cursor: 'DBCursor') -> None:
     )
 
 
+@enter_exit_debug_log()
+def set_unique_asset_collections(cursor: 'DBCursor') -> None:
+    """It does the following:
+    - Create a new asset_collections table with UNIQUE constraint and a temporary name.
+    - Copy all the asset collections to the new table, removing the duplicated ones.
+    - Delete the old table.
+    - Rename the new table to the old one.
+    """
+    v7_multiasset_mappings = cursor.execute(
+        'SELECT rowid, collection_id, asset FROM multiasset_mappings;',
+    ).fetchall()
+    cursor.execute("""
+        CREATE TABLE IF NOT EXISTS asset_collections_temp(
+            id INTEGER PRIMARY KEY,
+            name TEXT NOT NULL,
+            symbol TEXT NOT NULL,
+            UNIQUE (name, symbol)
+        );
+    """)
+
+    all_entries = cursor.execute('SELECT id, name, symbol FROM asset_collections;').fetchall()
+    unique_entries = set()
+    duplicated_entries = 0
+    for collection_id, name, symbol in all_entries:
+        if (name, symbol) not in unique_entries:
+            cursor.execute(
+                'INSERT INTO asset_collections_temp(id, name, symbol) VALUES (?, ?, ?);',
+                (collection_id - duplicated_entries, name, symbol),
```

Comment:
you need to explain with a comment what's this logic with "duplicated entries" here. I don't get it.